hooks/pre-checkout

```diff
@@ -89,14 +89,14 @@ export_buildkite_repo() {
     # when running aws cli v2, --cli-binary-format raw-in-base64-out is required
     aws lambda invoke \
         --region us-east-1 \
-        --function-name github-app-token \
+        --function-name github-app-token:stable \
```

Comment:
do you think it's a problem that the checkout plugin can start implicitly failing without changing anything here? I guess the error messages will always be explanatory enough, but there won't be a commit history here if this breaks due to the lambda breaking

Reply:
yeah this is the same concern we have with the checkout plugin and why we pin it to a specific commit hash instead of a tag, I think this won't be a problem because:
- There is always a log of what version of the lambda was called:
```

[2021-06-15 00:41:18] INFO: starting pre-checkout
--
 | {
 | "StatusCode": 200,
 | "ExecutedVersion": "$LATEST"
}
```
(`Latest` will be replaced for the actual version number once we use the alias)

- By using the alias we can do canary deployments and only promote a new version to `stable` once it has been tested and verified.